lib/ansible/runner.rb

```diff
@@ -24,13 +24,25 @@ def run_queue(env_vars, extra_vars, playbook_path, user_id, queue_opts)
       private
 
       def run_via_cli(env_vars, extra_vars, playbook_path)
-        result = AwesomeSpawn.run!(ansible_command, :env => env_vars, :params => [{:extra_vars => JSON.dump(extra_vars)}, playbook_path])
-        JSON.parse(result.output)
+        Dir.mktmpdir("ansible-runner") do |base_dir|
+          Dir.mkdir(File.join(base_dir, 'project')) # without this, there is a silent fail of the ansible-runner command see https://github.com/ansible/ansible-runner/issues/88
+
+          result = AwesomeSpawn.run!(ansible_command(base_dir), :env => env_vars, :params => [{:cmdline => "--extra-vars '#{JSON.dump(extra_vars)}'", :playbook => playbook_path}])
+
+          return Ansible::Runner::Response.new(:return_code => return_code(base_dir), :stdout => result.output, :stderr => result.error)
```

Comment:
Do you need the return here?  I think it will actually throw an exception:

```
>> a = Dir.mktmpdir("test") { |base_dir| return base_dir + "hello" }
LocalJumpError: unexpected return
```

Reply:
hm